hydromt/models/model_mesh.py

```diff
@@ -252,16 +292,135 @@ def set_mesh(
+                        ]
+                        # Re-define _mesh
+                        grids = xr.merge(grids)
+                        self._mesh = xu.UgridDataset(grids)
+            # Check again mesh_names, could have changed if overwrite_grid=True
+            if grid_name in self.mesh_names:
+                for dvar in data.data_vars:
+                    if dvar in self._mesh:
+                        self.logger.warning(f"Replacing mesh parameter: {dvar}")
+                    self._mesh[dvar] = data[dvar]
+            else:
+                # We are potentially adding a new grid without any data variables
+                self._mesh = xu.UgridDataset(
+                    xr.merge(
+                        [
+                            self.mesh.ugrid.to_dataset(optional_attributes=True),
+                            data.ugrid.to_dataset(optional_attributes=True),
+                        ]
+                    )
+                )
+            # Restore crs
+            for grid in self._mesh.ugrid.grids:
+                grid.set_crs(crs)
+
+        # update related geoms if necessary: region
+        if overwrite_grid or new_grid:
+            # add / updates region
+            self._geoms.pop("region", None)
+            self.region
+
+    def get_mesh(
+        self, grid_name: str, include_data: bool = False
+    ) -> Union[xu.UgridDataArray, xu.UgridDataset]:
+        """
+        Return a specific grid topology from mesh based on grid_name.
 
-    def read_mesh(self, fn: str = "mesh/mesh.nc", **kwargs) -> None:
+        If include_data is True, the data variables for that specific
+        grid are also included.
+
+        Parameters
+        ----------
+        grid_name : str
+            Name of the grid to return.
+        include_data : bool, optional
+            If True, also include data variables, by default False.
+
+        Returns
+        -------
+        uds: Union[xu.UgridDataArray, xu.UgridDataset]
+            Grid topology with or without data variables.
+        """
+        if self.mesh is None:
+            raise ValueError("Mesh is not set, please use set_mesh first.")
+        if grid_name not in self.mesh_names:
+            raise ValueError(f"Grid {grid_name} not found in mesh.")
+        if include_data:
+            grid = self.mesh_grids[grid_name]
+            uds = xu.UgridDataset(grid.to_dataset(optional_attributes=True))
+            uds.ugrid.grid.set_crs(grid.crs)
+            # Look for data_vars that are defined on grid_name
+            for var in self.mesh.data_vars:
+                if hasattr(self.mesh[var], "ugrid"):
+                    if self.mesh[var].ugrid.grid.name == grid_name:
+                        uds[var] = self.mesh[var]
+                # additionnal topology properties
+                elif var.startswith(grid_name):
+                    uds[var] = self.mesh[var]
+                # else is global property (not grid specific)
```

Comment:
do we need an else branch here? (not sure if the comment implies that we should or shouldn't)

Reply:
No no need for an else but I added the comment to kind of explain or be reminded why we don't need an else statement